### PR TITLE
Add documentation about the analyzer

### LINKED ISSUE #856: Document the analyzer
Nowhere do we mention that we have one.
We should at least put it in the docs.

### PR DESCRIPTION
**Pull request by thomaslevesque**
Fixes #856

### CODE REVIEW

File: docs/analyzer.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Analyzer
+
+FakeItEasy provides a Roslyn analyzer to detect incorrect usages of the library,
+which cannot be prevented by the API or the compiler but will typically result in
+bugs at runtime.
```
Comment (blairconrad) [inline docs/analyzer.md]: Grammar is hard, but I think ", which" should be "that".
As written, it sounds like _all_ incorrect usages of the library cannot be prevented by the API or the compiler, but I think the intent is for the analzyer to detect those incorrect usages that cannot be prevented by the API or compiler.

(If I'm not wrong, the second part of the sentence is a [nonrestrictive clause](http://www.quickanddirtytips.com/education/grammar/which-versus-that-0), and we want a restrictive one here.)

Perhaps I've misinterpreted your intent.

File: docs/analyzer.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Analyzer
+
+FakeItEasy provides a Roslyn analyzer to detect incorrect usages of the library,
+which cannot be prevented by the API or the compiler but will typically result in
+bugs at runtime.
+
+
```
Comment (blairconrad) [inline docs/analyzer.md]: Nit: double newline

File: docs/analyzer.md (hunk 1 above)
Comment (blairconrad) [inline docs/analyzer.md]: ", which" sounds to me like no incorrect usages of the library can be prevented by the API or the compiler.
I think we want a [restrictive clause](http://www.quickanddirtytips.com/education/grammar/which-versus-that-0) formed by "that", indicating the the analyzer is intended to help with the subset of incorrect usages that can't be prevented by the API or compiler.

Unless I've misinterpreted your intent?

**Comment by blairconrad** [comment]
(fun fact: the comment I thought I had to enter twice was actually saved and applied by GitHub. It just lied about having lost it.)

File: docs/analyzer.md (hunk 1 above)
Comment (thomaslevesque) [inline docs/analyzer.md]: Ah, yes,"that" sounds better. After 25 years of learning and speaking English, I *still* don't know when I should use "that" or "which"...

Comment (blairconrad) [inline docs/analyzer.md]: Nobody does.

**Comment by blairconrad** [comment]
Thanks, @thomaslevesque. Top-notch work.

**Comment by thomaslevesque** [comment]
Thanks for the merge!

File: docs/analyzer.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Analyzer
+
+FakeItEasy provides a Roslyn analyzer to detect incorrect usages of the library
+that cannot be prevented by the API or the compiler but will typically result in
+bugs at runtime.
```
Comment (adamralph) [inline docs/analyzer.md]: Should we be pre-empting word wrapping, or should we let readthedocs/the user's browser handle it?

Comment (blairconrad) [inline docs/analyzer.md]: Are you worried that this forces linebreaks?
I'm seeing different wrapping on readthedocs.

image

Comment (adamralph) [inline docs/analyzer.md]: Never mind, it seems like it's ignored. Even GitHub ignores it!

Comment (adamralph) [inline docs/analyzer.md]: The purist in me would remove them, but it seems they have no end effect either in the docs or in GitHub's rendering.

**Pull request merged**